### PR TITLE
[dev/eng] Start overhauling test build process.

### PR DESCRIPTION
**Pull request by mellinoe**
* Make System.Console.Tests build against the targeting pack.
* Make System.Runtime.Numerics.Tests build against the targeting pack.

This is just a small start but lays the groundwork for more projects.

@joperezr @karajas @weshaggard

### CODE REVIEW

File: targetingpacks.props [hunk 1]
```diff
@@ -12,4 +13,25 @@
...
+    <ItemGroup>
+      <!-- Whitelisted runtime assemblies that are OK to reference. -->
+      <RuntimeReferencedAssemblyNames Include="xunit.core" />
+      <RuntimeReferencedAssemblyNames Include="Xunit.NetCore.Extensions" />
+      <RuntimeReferencedAssemblyNames Include="xunit.assert" />
+      <RuntimeReferencedAssemblyNames Include="xunit.abstractions" />
+
+      <!-- Add Reference's to all files in the targeting pack folder, and to whitelisted items from the group above in the runtime folder. -->
+      <TargetingPackItems Include="%(TargetingPackDirs.Identity)/*" />
+      <Reference Include="%(TargetingPackItems.Filename)" Exclude="System.Private.CoreLib" /> <!-- TODO: System.Private.CoreLib shouldn't even be in the targeting pack. -->
```
Comment (weshaggard) [inline targetingpacks.props]: Do you know what is putting System.Private.CoreLib into the TP directory?

Comment (mellinoe) [inline targetingpacks.props]: This is: https://github.com/dotnet/corefx/tree/dev/eng/src/Microsoft.TargetingPack.Private.CoreCLR/ref

File: binplace.targets [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" InitialTargets="CheckForBuildTools" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <PropertyGroup>
-    <IsRuntimeAssembly Condition="'$(IsRuntimeAssembly)'=='' AND '$(IsReferenceAssembly)' != 'true'">true</IsRuntimeAssembly>
+    <IsRuntimeAssembly Condition="'$(IsRuntimeAssembly)'=='' AND '$(IsReferenceAssembly)' != 'true' AND '$(IsTestProject)' != 'true'">true</IsRuntimeAssembly>
```
Comment (joperezr) [inline binplace.targets]: Will this affect the RemoteExecutorConsoleApp.csproj as well? IsTestProject is calculated by looking if a project ends in Tests.csproj, so I wonder if that one would still get binplaced to the runtime folder.

File: src/Common/tests/System/Diagnostics/RemoteExecutorConsoleApp/RemoteExecutorConsoleApp.csproj [hunk 3]
```diff
@@ -8,7 +8,8 @@
     <AssemblyName>RemoteExecutorConsoleApp</AssemblyName>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <CopyNuGetImplementations>false</CopyNuGetImplementations>
-    <NugetTargetMoniker>.NETStandard,Version=v1.3</NugetTargetMoniker>
+    <IsTestProject>true</IsTestProject>
```
Comment (joperezr) [inline src/Common/tests/System/Diagnostics/RemoteExecutorConsoleApp/RemoteExecutorConsoleApp.csproj]: Oh, nvm, I see what you did here.

**Comment by joperezr** [comment]
>     <NugetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.7</NugetTargetMoniker>

Remove the `NugetTargetMoniker` as well

---
Refers to: src/System.Runtime.Numerics/tests/System.Runtime.Numerics.Tests.csproj:8 in ce574a8. [](commit_id = ce574a8cf158da02c0d6c9c6df5bc2d60d7c0108, deletion_comment = False)

File: src/System.Console/tests/System.Console.Tests.builds [hunk 4]
```diff
@@ -1,13 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
-  <ItemGroup>
-    <Project Include="System.Console.Tests.csproj" />
```
Comment (weshaggard) [inline src/System.Console/tests/System.Console.Tests.builds]: Before we start deleting builds files we need to be sure we capture the information in a BuildConfigurations property. @chcosta has generated some of them in his PR but I don't know if he has them all yet.

Comment (joperezr) [inline src/System.Console/tests/System.Console.Tests.builds]: I guess we don't really need to remove them yet, so in order to be able to merge without breaking @chcosta then we can just undo this delete.

**Comment by mellinoe** [comment]
Rebasing and then i'll merge after CI finishes. I have a bunch more changes on top of this to actually enable tests to start running, so I'll want to follow up with that stuff in a bit.

**Pull request merged**